src/site/content/en/blog/structured-clone/index.md

```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: Deep-copying in JavaScript using structuredClone
+authors:
+  - surma
+hero: image/i9nJGvw3SnTPH63zKOYWtI6cP5m2/e1QXRf64ylK8blZ45H1X.jpeg
+alt: Clones. Lots of clones.
+description: >
+  For the longest time, you had to resort to workarounds and libraries to create a deep copy of a JavaScript value. The Platform now ships with `structuredClone()`, a built-in function for deep-copying.
+date: 2021-12-13
+tags:
+  - performance
+  - blog
+---
+
+For the longest time, you had to resort to workarounds and libraries to create a deep copy of a JavaScript value. The Platform now ships with `structuredClone()`, a built-in function for deep-copying.
+
+{% BrowserCompat 'api.structuredClone' %}
+
+At the time of writing, _all browsers_ have already implemented this API in their nightly releases, Firefox has already shipped it to stable in Firefox 94. Additionally, Node 17 and Deno 1.14 implemented this API already as well. *You can start using this function right now* and not feel bad about it.
+
+## Shallow copies
+
+Copying a value in JavaScript is almost always _shallow_, as opposed to _deep_.  That means that changes to deeply nested values will be visible in the copy as well as the original. 
+
+```js
+const myOriginal = {
+  someProp: "with a string value",
+  anotherProp: {
+    withAnotherProp: 1,
+    andAnotherProp: true
+  }
+};
+
+// Object spread is one way to create a shallow copy in JS.
+const myShallowCopy = {...myOriginal};
+
+// Adding or changing a property to the shallow copy...
+myShallowCopy.aNewProp = "a new value";
+// ... will only affect the copy
+console.log(myOriginal.aNewProp)
+// ^ logs `undefined`
+
+
+// While adding or changing a deeply nested property...
+myShallowCopy.anotherProp.aNewProp = "a new value";
+// ... will affect the original as well
+console.log(myOriginal.anotherProp.aNewProp) 
+// ^ logs `a new value`
+```
+
+The expression `{...myOriginal}` iterates over the (enumerable) properties of `myOriginal` using the [Spread Operator]. It uses the property name and value, and assigns them one by one to a freshly created, empty object. As such, the resulting object is identical in shape, but with its own copy of the list of properties and values. The values are copied, too, but so-called primitive values are handled differently by the JavaScript value than non-primitive values. To quote [MDN][MDN Primitive]:
```

Comment:
Looking at this in the preview I think it might be clearer if it were broken down into three js blocks with an explanation before each rather than replying on code comments.

Reply:
Fair. Split the code samples apart. PTAL.